tests/checks/common.py

```diff
@@ -497,7 +494,49 @@ def assertWarning(self, warning, count=None, at_least=1, exact_match=True):
             self._candidates_size_assert(candidates, count=count, at_least=at_least)
         except AssertionError:
             log.error("Candidates size assertion for {0}, count: {1}, "
-                "at_least: {2}) failed".format(warning, count, at_least))
+                      "at_least: {2}) failed".format(warning, count, at_least))
             raise
 
         log.debug("{0} FOUND !".format(warning))
+
+    # Potential kwargs: aggregation_key, alert_type, event_type,
+    # msg_title, source_type_name
+    def assertEvent(self, msg_text, count=None, at_least=1, exact_match=True,
+                    tags=None, **kwargs):
+        log.debug("Looking for event {0}".format(msg_text))
+        if tags is not None:
+            log.debug(" * tagged with {0}".format(tags))
+        for name, value in kwargs.iteritems():
+            if value is not None:
+                log.debug(" * with {0} {1}".format(name, value))
+        if count is not None:
+            log.debug(" * should have exactly {0} events".format(count))
+        elif at_least is not None:
+            log.debug(" * should have at least {0} events".format(count))
+
+        candidates = []
+        for e in self.events:
+            if exact_match and msg_text != e['msg_text'] or \
+                    not exact_match and msg_text not in e['msg_text']:
+                continue
+            if tags and set(tags) != set(e['tags']):
+                continue
+            for name, value in kwargs.iteritems():
+                if e[name] != value:
+                    break
+            else:
+                candidates.append(e)
+
+        try:
+            self._candidates_size_assert(candidates, count=count, at_least=at_least)
+        except AssertionError:
+            log.error("Candidates size assertion for {0}, count: {1}, "
+                      "at_least: {2}) failed".format(msg_text, count, at_least))
+            raise
+
+        for ev in self.events:
```

Comment:
Completely on a side note, as I know you like 'beauteous' code.
I really like `itertools.product` to write nested loops in one line, in case you don't know it:
```python
from itertools import product

for ev, ec in product(self.events, candidates):
    if ec == ev:
        pass
```